onnxruntime/core/framework/allocatormgr.h

```diff
@@ -11,21 +11,30 @@ namespace onnxruntime {
 
 using DeviceAllocatorFactory = std::function<std::unique_ptr<IDeviceAllocator>(OrtDevice::DeviceId)>;
 
+// TODO why does DeviceAllocatorRegistrationInfo have arena related configs?
+// TODO even if it should, they should be inside their own struct (OrtArenaCfg) as opposed to
```

Comment:
I agree. Arena is a layer above the device allocator, so the arena extend strategy seems to be at the wrong level. Alternatively 'DeviceAllocatorRegistrationInfo' is a bad name and it needs to be something more generic and less device specific.

Reply:
Changing this will create a lot of churn in this PR and it's kinda unrelated. Will address in a separate PR. Added work item https://msdata.visualstudio.com/Vienna/_workitems/edit/863400.